crates/holochain_p2p/src/hc_report.rs

```diff
@@ -0,0 +1,599 @@
+use holochain_types::report::{ReportEntry, ReportEntryFetchedOps};
+use kitsune2_api::*;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex, Weak};
+
+const MOD: &str = "HCREPORT";
+
+/// HcReport configuration types.
+mod config {
+    /// Configuration parameters for [HcReportFactory](super::HcReportFactory).
+    #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct HcReportConfig {
+        /// How many days worth of report files to retain.
+        pub days_retained: u32,
+
+        /// Directory path for the report files.
+        /// Files will be named `hc-report.YYYY-MM-DD.jsonl`.
+        pub path: std::path::PathBuf,
+
+        /// How often to report Fetched-Op aggregated data in seconds.
+        pub fetched_op_interval_s: f64,
+    }
+
+    impl Default for HcReportConfig {
+        fn default() -> Self {
+            Self {
+                days_retained: 5,
+                path: "/tmp/hc-report".into(),
```

Comment:
Perhaps a code comment that this is always provided by Holochain to place reports alongside other conductor data?